Userland/Libraries/LibWeb/HTML/StructuredSerialize.h

```diff
@@ -61,8 +61,13 @@ WebIDL::ExceptionOr<void> serialize_big_int_object(JS::VM& vm, SerializationReco
 WebIDL::ExceptionOr<void> serialize_string_object(JS::VM& vm, SerializationRecord& serialized, JS::Value& value);
 void serialize_date_object(SerializationRecord& serialized, JS::Value& value);
 WebIDL::ExceptionOr<void> serialize_reg_exp_object(JS::VM& vm, SerializationRecord& serialized, JS::Value& value);
-void serialize_u64(SerializationRecord& serialized, u64 value);
-void serialize_i64(SerializationRecord& serialized, i64 value);
+
+template<typename T>
+requires(IsIntegral<T> || IsFloatingPoint<T>)
+void serialize_primitive_type(SerializationRecord& serialized, T value)
+{
+    serialized.append(bit_cast<u32*>(&value), sizeof(value) / 4);
+}
```

Comment:
Realizing this and `deserialize_primitive_type()` won't work since they will accept values not divisible by 4.

Reply:
I think you would need to encode some promotion logic in here. 

```
if constexpr (sizeof(T) < sizeof(u32)) {
    u32 promoted_value = static_cast<u32>(value);
    ....
}
```

Reply:
on the other end it gets a bit awkward, because if we want to avoid UB we need to make sure that the value of `deserialized_promoted_value` will actually fit into the bounds of T before casting.

Reply:
... because I like sending emails, here's another thought: would be neato to have a `serialize_enum` that defers to `serialize_primitive_type<Underlying<T>>(to_underlying(value));`

Reply:
Thanks, Andrew! Added the promotion logic as suggested, also added `serialize_enum()`.